src/components/themed/MenuTabs.tsx

```diff
@@ -63,9 +69,35 @@ export const MenuTabs = (props: BottomTabBarProps) => {
       case 'walletsTab':
         return navigation.navigate('walletsTab', currentName === 'walletsTab' ? { screen: 'walletList' } : {})
       case 'buyTab':
-        return navigation.navigate('buyTab', currentName === 'buyTab' ? { screen: 'pluginListBuy' } : {})
+        if (isLightAccount) {
+          Airship.show((bridge: AirshipBridge<BackupForTransferModalResult | undefined>) => {
+            return <BackupForTransferModal bridge={bridge} />
+          })
+            .then((userSel?: BackupForTransferModalResult) => {
+              if (userSel === 'upgrade') {
+                navigation.navigate('upgradeUsername', {})
+              }
+            })
+            .catch(error => showError(error))
```

Comment:
I wonder if you could make a helper function, so it's more like:

```ts
 await showUpgradeModal(() => navigation.navigate('upgradeUsername', {}))
```

That way 90% of the code vanishes, and just the `navigation` stuff has to be duplicated.

Reply:
Ah yeah, I couldn't get it work earlier because I was trying to take a navigation as a prop directly